### PR TITLE
[OBMIN-61] Fixed refresh page bug, refactored Controller and Flow tests

### PR DESCRIPTION
**Pull request by rpkyrych**

### CODE REVIEW

File: src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java [hunk 1]
```diff
@@ -297,40 +233,27 @@ void selfDeleteRequest_WhenPasswordWrongAndTheSameConfirmation_ShouldThrowInvali
...
-                status().isBadRequest())
-                .andDo(print())
-                .andReturn();
-        String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
-
-        assertTrue(message.contains(getMessageSource("different.passwords")));
+    private static Stream<Arguments> selfDeleteTestData() {
+        return Stream.of(
+                Arguments.of(new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD)),
+                Arguments.of(new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD))
```
Comment (Jack11M) [inline src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java:255]: Why do we need the same data here (new UserDeleteFlowDto(CORRECT_OLD_PASSWORD, WRONG_OLD_PASSWORD)) ?

File: src/test/java/space/obminyashka/items_exchange/end2end/LocationFlowTest.java [hunk 2]
```diff
@@ -68,168 +70,84 @@ void getLocationsForCurrentLanguage_shouldReturnLocations() throws Exception {
...
     void getLocation_shouldReturn404WhenLocationIsNotExisted() throws Exception {
-        mockMvc.perform(get("/location/{location_id}", 555L))
-                .andDo(print())
-                .andExpect(status().isNotFound())
-                .andReturn();
-    }
-
-    @WithMockUser(username = "admin")
-    @Test
-    void createLocation_shouldReturn409WhenUserHasNotRoleAdmin() throws Exception {
```
Comment (Jack11M) [inline src/test/java/space/obminyashka/items_exchange/end2end/LocationFlowTest.java:97]: Why have you deleted all tests to check the ADMIN role?

Comment (Jack11M) [reply src/test/java/space/obminyashka/items_exchange/end2end/LocationFlowTest.java:97]: Sorry, I understood. We do not have this functionality)

**Review by Jack11M — APPROVED** [review]

**Comment by Jack11M** [comment]
Great!!! It`s magic!

File: src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java (hunk 1 above)
Comment (rpkyrych) [reply src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java:255]: 😄 two tests which existed before had the same values and actually did the same, I only refactored and was pretty tired to find the difference into the data. Will be changed, thanks.

**Review by Basist92 — APPROVED** [review]

**Pull request merged**